diff --git a/src/video/cocoa/SDL_cocoawindow.m b/src/video/cocoa/SDL_cocoawindow.m
index a878267..76e5ff7 100644
--- a/src/video/cocoa/SDL_cocoawindow.m
+++ b/src/video/cocoa/SDL_cocoawindow.m
@@ -2504,6 +2504,7 @@ bool Cocoa_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window, SDL_Properti
             contentView = [[SDL3View alloc] initWithFrame:rect];
             [contentView setSDLWindow:window];
             nsview = contentView;
+            [nswindow setContentView:nsview];
         }
 
         if (window->flags & SDL_WINDOW_ALWAYS_ON_TOP) {
@@ -2542,8 +2543,6 @@ bool Cocoa_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window, SDL_Properti
         }
 #endif // SDL_VIDEO_OPENGL_EGL
 #endif // SDL_VIDEO_OPENGL_ES2
-        [nswindow setContentView:nsview];
-
         if (!SetupWindowData(_this, window, nswindow, nsview)) {
             return false;
         }
